pdg/src/snapshots/c2rust_pdg__tests__analysis_test_pdg_snapshot.snap

```diff
@@ -488,18 +498,21 @@ g {
 	n[3]:  value.store n[2]  => _5.*.0 @ bb4[7]:  fn lighttpd_test;     
 	n[4]:  value.load  _     => _8     @ bb0[3]:  fn fdevent_register;  
 	n[5]:  offset[0]   n[4]  => _7     @ bb0[8]:  fn fdevent_register;  
-	n[6]:  addr.store  n[5]  => _      @ bb2[0]:  fn fdevent_register;  
-	n[7]:  value.load  _     => _5     @ bb0[4]:  fn fdevent_unregister;
-	n[8]:  offset[0]   n[7]  => _4     @ bb0[9]:  fn fdevent_unregister;
-	n[9]:  addr.load   n[8]  => _      @ bb1[2]:  fn fdevent_unregister;
-	n[10]: value.load  _     => _19    @ bb7[5]:  fn fdevent_unregister;
-	n[11]: offset[0]   n[10] => _18    @ bb7[10]: fn fdevent_unregister;
-	n[12]: addr.store  n[11] => _      @ bb8[4]:  fn fdevent_unregister;
-	n[13]: copy        n[1]  => _20    @ bb6[7]:  fn lighttpd_test;     
-	n[14]: copy        n[13] => _19    @ bb6[8]:  fn lighttpd_test;     
-	n[15]: free        n[14] => _18    @ bb6[9]:  fn lighttpd_test;     
-}
-nodes_that_need_write = [12, 11, 10, 6, 5, 4]
+	n[6]:  copy        n[5]  => _6     @ bb1[3]:  fn fdevent_register;  
+	n[7]:  addr.store  n[5]  => _      @ bb2[0]:  fn fdevent_register;  
+	n[8]:  addr.load   n[5]  => _      @ bb2[3]:  fn fdevent_register;  
```

Comment:
to clarify, `fresh0` is a reference, which is why this is new

Reply:
Looking at this and the previous bit, if the code is something like

```Rust
let ref mut fresh0 = *((*ev).fdarray).offset(fd as isize);
*fresh0 = ...;   // Is this present?  bb2[0]?
let mut fdn: *mut fdnode = *fresh0;  // bb2[3]?
```

then I think `n[7]` and `n[8]` should have `n[6]` (`_6` / `fresh0`) as their source, rather than `n[5]` (`_7` / the temporary storing the result of `offset`).  Also, if the original Rust code doesn't contain a write to `*fresh0` like the line I marked `bb2[0]?`, then `n[7]` should not be present - there should only be a `StoreAddr` if `*fresh0` is actually written, and things like `&mut (*fresh0)` don't actually touch memory.

Reply:
regarding the source node of `n[6]`, I think you're right and this is a bug. looking into it

Reply:
Fixed in 3f4b5a8e